File: db/postgres/gitlabMergeRequest.js
'use strict'

const {
  Model
} = require('sequelize')

module.exports = (sequelize, DataTypes) => {
  class GitlabMergeRequest extends Model {

  }

  GitlabMergeRequest.init({
    id: {
      primaryKey: true,
      type: DataTypes.INTEGER
    },
    title: {
      type: DataTypes.STRING
    },
    projectId: {
      type: DataTypes.INTEGER
    },
    numberOfReviewers: {
      type: DataTypes.INTEGER
    },
    state: {
      type: DataTypes.STRING
    },
    webUrl: {
      type: DataTypes.STRING
    },
    userNotesCount: {
      type: DataTypes.INTEGER
    },
    workInProgress: {
      type: DataTypes.BOOLEAN
    },
    sourceBranch: {
      type: DataTypes.STRING
    },
    mergedAt: {
      type: DataTypes.DATE
    },
    gitlabCreatedAt: {
      type: DataTypes.DATE
    },
    closedAt: {
      type: DataTypes.DATE
    },
    mergedByUsername: {
      type: DataTypes.STRING
    },
    description: {
      type: DataTypes.STRING
    },
    authorUsername: {
      type: DataTypes.STRING
    },
    createdAt: {
      allowNull: false,
      type: DataTypes.DATE,
      defaultValue: DataTypes.NOW
    },
    updatedAt: {
      allowNull: false,
      type: DataTypes.DATE,
      defaultValue: DataTypes.NOW
    }
  }, {
    sequelize,
    modelName: 'GitlabMergeRequest',
    underscored: true
  })

  return GitlabMergeRequest
}

File: plugins/gitlab/models/gitlab_merge_request.go
package models

import (
	"time"

	"github.com/merico-dev/lake/models/common"
)

type GitlabMergeRequest struct {
	GitlabId         int `gorm:"primaryKey"`
	Iid              int `gorm:"index"`
	ProjectId        int `gorm:"index"`
	SourceProjectId  int
	TargetProjectId  int
	State            string `gorm:"type:varchar(255)"`
	Title            string
	WebUrl           string `gorm:"type:varchar(255)"`
	UserNotesCount   int
	WorkInProgress   bool
	SourceBranch     string `gorm:"type:varchar(255)"`
	TargetBranch     string `gorm:"type:varchar(255)"`
	MergeCommitSha   string `gorm:"type:varchar(255)"`
	MergedAt         *time.Time
	GitlabCreatedAt  time.Time
	ClosedAt         *time.Time
	MergedByUsername string `gorm:"type:varchar(255)"`
	Description      string
	AuthorUsername   string `gorm:"type:varchar(255)"`
	AuthorUserId     int
	Component        string     `gorm:"type:varchar(255)"`
	FirstCommentTime *time.Time `gorm:"comment:Time when the first comment occurred"`
	ReviewRounds     int        `gorm:"comment:How many rounds of review this MR went through"`
	common.NoPKModel
}

func (GitlabMergeRequest) TableName() string {
	return "_tool_gitlab_merge_requests"
}
